packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m

```diff
@@ -153,11 +161,9 @@ - (void)signIn:(GIDSignIn *)signIn
 #pragma mark - private methods
 
 - (void)respondWithAccount:(id)account error:(NSError *)error {
-  NSArray<FlutterResult> *requests = _accountRequests;
-  _accountRequests = [[NSMutableArray alloc] init];
-  for (FlutterResult accountRequest in requests) {
-    accountRequest(error != nil ? error.flutterError : account);
-  }
+  FlutterResult result = _accountRequest;
```

Comment:
Might cause a race condition? _accountRequest() signals the app to send a platform message back indicating the completion of the operation. Plugin itself does not consider the operation complete until _accountRequest is nil.

It is safer to mark the operation as completed on the plugin side _then_ signal the app.

I am not sure if this is a realistic concern.